src/test/groovy/graphql/introspection/IntrospectionTest.groovy

```diff
@@ -57,4 +64,137 @@ class IntrospectionTest extends Specification {
                 ]
         ]
     }
+
+    def "can query directives on types via graphql java specific api"() {
+        given:
+        def spec = '''
+            directive @myDirective(myArg: String) on FIELD_DEFINITION | OBJECT
+           
+            type Query @myDirective(myArg: "on query") {
+                hello : String @myDirective(myArg: "on field")
+            }
+        '''
+        def schema = TestUtil.schema(spec)
+        def schemaTransformer = new SchemaTransformer()
+        def schemaWithExposedDirective = schemaTransformer.transform(schema, new GraphQLTypeVisitorStub() {
+
+            @Override
+            TraversalControl visitGraphQLDirective(GraphQLDirective graphQLDirective, TraverserContext<GraphQLSchemaElement> context) {
+                def newDirective = graphQLDirective.transform({ it.exposeViaIntrospection(true) })
+                return TreeTransformerUtil.changeNode(context, newDirective)
+            }
+        })
+
+        def graphql = GraphQL.newGraphQL(schemaWithExposedDirective).build()
+
+        def query = """
+        { __schema {
+                types {
+                ...FullType
+                }
+            }
+       }
+      fragment FullType on __Type  {
+        directives_GJ_API {
```

Comment:
Naming is hard but I really hate this naming convention.

My thoughts are

* __gqljava_directives
* ___directives   (triple underscores - eg like the standard double underscoeres but even more specific)